mailpoet/assets/js/src/automation/templates/components/template-detail.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  ComponentType,
+  forwardRef,
+  MouseEventHandler,
+  ReactNode,
+  useCallback,
+  useState,
+} from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { Button, Modal, Snackbar as WpSnackbar } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { chevronLeft, chevronRight, warning } from '@wordpress/icons';
+import { Tag } from '@woocommerce/components';
+import { addQueryArgs } from '@wordpress/url';
+import { TemplatePreview } from './template-preview';
+import { AutomationTemplate, automationTemplateCategories } from '../config';
+import { MailPoet } from '../../../mailpoet';
+
+// snackbar icon is not annotated in the types
+const Snackbar = WpSnackbar as ComponentType<
+  WpSnackbar.Props & {
+    icon: ReactNode;
+  }
+>;
+
+const getCategory = (template: AutomationTemplate): string =>
+  automationTemplateCategories.find(({ slug }) => slug === template.category)
+    ?.name ?? __('Uncategorized', 'mailpoet');
+
+const useCreateFromTemplate = () => {
+  const [state, setState] = useState({
+    data: undefined,
+    loading: false,
+    error: undefined,
+  });
+
+  const create = useCallback(async (slug: string) => {
+    setState((prevState) => ({ ...prevState, loading: true }));
+    try {
+      const data = await apiFetch<{ data: { id: string } }>({
+        path: `/automations/create-from-template`,
+        method: 'POST',
+        data: { slug },
+      });
+      MailPoet.trackEvent('Automations > Template selected', {
+        'Automation slug': slug,
+      });
+      window.location.href = addQueryArgs(MailPoet.urls.automationEditor, {
+        id: data.data.id,
+      });
+    } catch (error) {
+      setState((prevState) => ({ ...prevState, error }));
+    } finally {
+      setState((prevState) => ({ ...prevState, loading: false }));
+    }
+  }, []);
+
+  return [create, state] as const;
+};
+
+type Props = {
+  template: AutomationTemplate;
+  onRequestClose: Modal.Props['onRequestClose'];
+  onPreviousClick?: MouseEventHandler<HTMLButtonElement>;
+  onNextClick?: MouseEventHandler<HTMLButtonElement>;
+};
+
+export const TemplateDetail = forwardRef<HTMLDivElement, Props>(
+  ({ template, onRequestClose, onPreviousClick, onNextClick }, ref) => {
+    const [createAutomationFromTemplate, { loading, error }] =
+      useCreateFromTemplate();
+
+    return (
+      <Modal
+        ref={ref}
+        className="mailpoet-automation-template-detail"
+        title=""
+        onRequestClose={onRequestClose}
+      >
+        <div className="mailpoet-automation-template-detail-content">
+          <div className="mailpoet-automation-template-detail-info">
+            <Tag label={getCategory(template)} />
+            <h1>{template.name}</h1>
+            {template.description}
+          </div>
+          <div className="mailpoet-automation-template-detail-preview">
```

Comment:
If a given automation has a filter, clicking on the label "Filters" in the preview causes the site to load an empty page.

Reply:
The modal seems too small for the user to view its contents. I'm testing using Chrome. Do you get something different?

![Screenshot from 2023-12-01 11-35-30](https://github.com/mailpoet/mailpoet/assets/77215/d53af7d8-3cac-4908-9ea4-0948926c59b3)

In the browser console, I see:

```
react-dom.development.js:86 Warning: validateDOMNesting(...): <button> cannot appear as a descendant of <button>.
    at button
    at UnforwardedButton (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/button/index.js:114:7)
    at FilterWrapper (webpack-internal:///./assets/js/src/automation/editor/components/filter-wrapper/index.tsx:28:5)
    at div
    at div
    at div
    at div
    at FiltersList (webpack-internal:///./assets/js/src/automation/editor/components/filters/list.tsx:36:5)
    at div
    at div
    at div
    at VisualElementHandler (webpack-internal:///../node_modules/.pnpm/framer-motion@7.10.3_react-dom@18.2.0_react@18.2.0/node_modules/framer-motion/dist/es/motion/utils/VisualElementHandler.mjs:8:1)
    at MotionComponent (webpack-internal:///../node_modules/.pnpm/framer-motion@7.10.3_react-dom@18.2.0_react@18.2.0/node_modules/framer-motion/dist/es/motion/index.mjs:50:65)
    at eval (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/popover/index.js:86:12)
    at UnforwardedPopover (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/popover/index.js:122:5)
    at div
    at UnforwardedDropdown (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/dropdown/index.js:46:5)
    at FiltersChip (webpack-internal:///./assets/js/src/automation/editor/components/filters/chip.tsx:25:5)
    at div
    at button
    at Comp (webpack-internal:///../node_modules/.pnpm/reakit-system@0.15.2_react-dom@18.2.0_react@18.2.0/node_modules/reakit-system/es/createComponent.js:50:26)
    at div
    at Step (webpack-internal:///./assets/js/src/automation/editor/components/automation/step.tsx:57:5)
    at div
    at StepWrapper (webpack-internal:///./assets/js/src/automation/editor/step-wrapper.tsx:25:5)
    at FlowStep (webpack-internal:///./assets/js/src/automation/editor/components/automation/flow-step.tsx:26:56)
    at div
    at div
    at Flow (webpack-internal:///./assets/js/src/automation/editor/components/automation/flow.tsx:21:5)
    at div
    at div
    at div
    at Comp (webpack-internal:///../node_modules/.pnpm/reakit-system@0.15.2_react-dom@18.2.0_react@18.2.0/node_modules/reakit-system/es/createComponent.js:50:26)
    at Automation (webpack-internal:///./assets/js/src/automation/editor/components/automation/index.tsx:38:5)
    at TemplatePreview (webpack-internal:///./assets/js/src/automation/templates/components/template-preview.tsx:38:5)
    at div
    at div
    at div
    at div
    at div
    at StyleProvider (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/style-provider/index.js:43:5)
    at div
    at UnforwardedModal (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/modal/index.js:51:5)
    at eval (webpack-internal:///./assets/js/src/automation/templates/components/template-detail.tsx:87:5)
    at TemplatesGrid (webpack-internal:///./assets/js/src/automation/templates/components/templates-grid.tsx:37:5)
    at div
    at div
    at TabPanel (webpack-internal:///../node_modules/.pnpm/@wordpress+components@23.5.0_@types+react@18.0.28_react-dom@18.2.0_react@18.2.0/node_modules/@wordpress/components/build-module/tab-panel/index.js:90:5)
    at TabPanel
    at div
    at Templates
```